hedera-mirror-rest/viewmodel/contractResultDetailsViewModel.js

```diff
@@ -64,6 +64,17 @@ class ContractResultDetailsViewModel extends ContractResultViewModel {
         ? ContractResultDetailsViewModel._SUCCESS_RESULT
         : ContractResultDetailsViewModel._FAIL_RESULT;
 
+    if (this.status === ContractResultDetailsViewModel._FAIL_RESULT) {
+      this.failed_initcode = !_.isEmpty(contractResult.failedInitcode)
+        ? utils.toHexStringNonQuantity(contractResult.failedInitcode)
+        : null;
+      if (_.isNil(this.failed_initcode) && !_.isEmpty(transaction.callData)) {
+        this.failed_initcode = utils.toHexStringNonQuantity(transaction.callData);
+      }
+    } else {
+      this.failed_initcode = null;
+    }
```

Comment:
nit: Could maybe simplify the conditionals:

```suggestion
    if (!_.isEmpty(contractResult.failedInitcode)) {
      this.failed_initcode = utils.toHexStringNonQuantity(contractResult.failedInitcode);
    } else if (this.status === ContractResultDetailsViewModel._FAIL_RESULT && !_.isEmpty(transaction.callData)) {
      this.failed_initcode = utils.toHexStringNonQuantity(transaction.callData);
    }
```

Reply:
Thanks I have updated the conditional, though I still had one part at the end to set the failed_initcode to null.